CHANGELOG.md

```diff
@@ -2,57 +2,213 @@
+        - `UnmarshalBinary([]byte) (error)`: standard deserialization.
+    - Streamlined and simplified all tests related to serialization. They can now be implemented with a single line of code with `RequireSerializerCorrect` which checks the correctness of all the above interface as well as equality between bites writen using `WriteTo` and bytes generated using `MarshalBinary`.
+- Improved consistency across method names:
+    - All sub-strings `NoMod`, `NoModDown` and `Constant` in methods names have been replaced by the sub-string `Lazy`. For example `AddNoMod` and `MulCoeffsMontgomeryConstant` become `AddLazy` and `MulCoeffsMontgomeryLazy` respectively.
+    - All sub-strings `And` in methods names have been replaced by the sub-string `Then`. For example `MulAndAdd` becomes `MulThenAdd`.
+    - All sub-strings `Inv` have been replaced by `I` for consistency. For example `InvNTT` becomes `INTT`.
+    - All sub-strings `Params` and alike referring to pre-computed constants have been replaced by `Constant`. For example `ModUpParams` becomes `ModUpConstants`.
+- New Packages:
+    - `he`: Package `he` implements scheme agnostic functionalities for the RLWE-based HE schemes implemented in Lattigo.
+        - Linear Transformations
+        - Polynomial Evaluation
+    - `he/hefloat`: Package `hefloat` implements fixed-point approximate encrypted arithmetic over reals/complex numbers. This package provides all the functionalities of the `schemes/ckks` package, as well as additional more advanced circuits, such as:
+        - Linear Transformations
+        - Homomorphic encoding/decoding
+        - Polynomial Evaluation
+        - Composite Minimax Polynomial Evaluation
+        - Homomorphic modular reduction (x mod 1)
+        - GoldschmidtDivision (x in [0, 2])
+        - Full domain division (x in [-max, -min] U [min, max])
+        - Sign and Step piece-wise functions (x in [-1, 1] and [0, 1] respectively)
+        - Min/Max between values in [-0.5, 0.5]
+    - `he/hefloat/bootstrapper`: Package `bootstrapper` implements a bootstrapping helper above the package `he/hefloat/bootstrapper/bootstrapping`. It notably enables:
+        - Bootstrapping batches of ciphertexts of smaller dimension and/or with sparse packing with ring-degree switching and depth-less packing/unpacking.
+        - Bootstrapping for the Conjugate Invariant CKKS with optimal throughput.
+    - `he/hefloat/bootstrapper/bootstrapping`: Package `bootstrapping` implements the core of the bootstrapping for approximate homomorphic encryption with a very parameterization granularity.
+        - Decorelation between the bootstrapping parameters and residual parameters: the user doesn't need to manage two sets of parameters anymore and the user only needs to provide the residual parameters (what should remains after the evaluation of the bootstrapping circuit)
+        - Right out of the box usability with default parameterization independent of the residual parameters
```

Comment:
The two lines above are confusing. I did not rephrased them before we decide if two packages for the bootstrapping are necessary.

Reply:
Merged both packages